app/workers/csv_export_worker.rb

```diff
@@ -0,0 +1,57 @@
+require 'fog/aws'
+
+class CsvExportWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: 0
+  sidekiq_options queue: 'export_csv'
+
+  def perform(search_params, recipient_address)
+    search_params = search_params.symbolize_keys
+    presenter = get_csv_presenter(search_params)
+
+    basename = presenter.filename
+    csv_string = presenter.csv_rows
+
+    file_url = upload_to_s3("#{basename}.csv", csv_string)
+
+    # Send email with link
+    ContentCsvMailer.content_csv_email(recipient_address, file_url).deliver_now
+  end
+
+  def get_csv_presenter(search_params)
+    document_types = FetchDocumentTypes.call[:document_types]
+    organisations = FetchOrganisations.call[:organisations]
+
+    content_items = FindContent.enum(search_params)
+    ContentItemsCSVPresenter.new(
+      content_items,
+      search_params,
+      document_types,
+      organisations
+    )
+  end
+
+  def upload_to_s3(filename, body)
```

Comment:
Uploading a file to S3 seems like a concern that should not live in the job. Why not just extracting a class with this behaviour and reuse it from this class? Testing would also look better.

Reply:
I've moved to being a concern